internal/auth/telemetry_claim.go

```diff
@@ -51,3 +53,36 @@ func getTelemetryClaim(ctx context.Context) (*TelemetryClaim, error) {
 	}
 	return claim, nil
 }
+
+func ValidAttestationClaim(ctx context.Context, filter *model.AttestationFilter) bool {
+	claim, err := getTelemetryClaim(ctx)
+	if err != nil || claim.CloudEvents == nil {
+		return false
+	}
+
+	source := tokenclaims.GlobalIdentifier
+	id := tokenclaims.GlobalIdentifier
+
+	if filter != nil {
+		if filter.Source != nil {
+			source = filter.Source.Hex()
+		}
+
+		if filter.ID != nil {
+			id = *filter.ID
+		}
+	}
+
+	var validClaim bool
+	for _, ce := range claim.CloudEvents.Events {
+		if ce.EventType == cloudevent.TypeAttestation || ce.EventType == tokenclaims.GlobalIdentifier {
+			if ce.Source == source || ce.Source == tokenclaims.GlobalIdentifier {
+				if slices.Contains(ce.IDs, id) || slices.Contains(ce.IDs, tokenclaims.GlobalIdentifier) {
+					validClaim = true
```

Comment:
Maybe `return true` here and `return false` at the end of the function?

Reply:
updated!